x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_signal.ts

```diff
@@ -5,35 +5,29 @@
  */
 
 import { RulesSchema } from '../../../../common/detection_engine/schemas/response/rules_schema';
-import { INTERNAL_IDENTIFIER } from '../../../../common/constants';
 import { SignalSourceHit, Signal, Ancestor } from './types';
 
-export const buildAncestor = (doc: SignalSourceHit, rule: Partial<RulesSchema>): Ancestor => {
-  const existingSignal = doc._source.signal?.parent;
-  if (existingSignal != null) {
+export const buildParent = (doc: SignalSourceHit): Ancestor => {
+  if (doc._source.signal != null) {
     return {
-      rule: rule.id != null ? rule.id : '',
+      rule: doc._source.signal.rule.id,
```

Comment:
Semantics change: this now populates `signal.parents.rule` now contains the parent rule rather than current signal rule.